socorro/processor/rules/breakpad.py

```diff
@@ -63,188 +60,73 @@ def action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-    # no effect on anything and are just here.
-    required_config = Namespace()
-    required_config.add_option(
-        'dump_field',
-        doc='the default name of a dump',
-        default='upload_file_minidump',
-    )
-    required_config.add_option(
-        'command_line',
-        doc='template for the command to invoke the external program; uses Python format syntax',
-        default='timeout -s KILL 30 {command_pathname}',
-    )
-    required_config.add_option(
-        'command_pathname',
-        doc='the full pathname to the external program to run (quote path with embedded spaces)',
-        default='/data/socorro/stackwalk/bin/dumplookup',
-    )
-    required_config.add_option(
-        'result_key',
-        doc='where the external process result should be stored in the processed crash',
-        default='%s_result' % (
-            required_config.command_pathname.default.split('/')[-1].replace('-', '')
-        ),
-    )
-    required_config.add_option(
-        'return_code_key',
-        doc='where the external process return code should be stored in the processed crash',
-        default='%s_return_code' % required_config.command_pathname.default.split('/')[-1],
-    )
-
-    def __init__(self, config):
-        super().__init__(config)
+def dot_save(a_mapping, key, value):
+    if '.' not in key or isinstance(a_mapping, DotDict):
+        a_mapping[key] = value
+        return
 
-    def _interpret_external_command_output(self, fp, processor_meta):
-        data = fp.read()
+    # FIXME(willkg): Replace this with glom
+    current_mapping = a_mapping
+    key_parts = key.split('.')
+    for key_fragment in key_parts[:-1]:
         try:
-            return json.loads(data)
-        except Exception as x:
-            self.logger.error(
-                '%s non-json output: "%s"' % (self.config.command_pathname, data[:100])
-            )
-            processor_meta.processor_notes.append(
-                "%s output failed in json: %s" % (self.config.command_pathname, x)
-            )
-        return {}
-
-    def _execute_external_process(self, crash_id, command_line, processor_meta):
-        # Tokenize the command line into args
-        command_line_args = shlex.split(command_line, comments=False, posix=True)
-
-        # Execute the command line sending stderr (debug logging) to devnull and
-        # capturing stdout (JSON blob of output)
-        subprocess_handle = subprocess.Popen(
-            command_line_args,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.DEVNULL,
+            current_mapping = current_mapping[key_fragment]
+        except KeyError:
+            current_mapping[key_fragment] = {}
+            current_mapping = current_mapping[key_fragment]
+    current_mapping[key_parts[-1]] = value
+
+
+def execute_external_process(crash_id, command_pathname, command_line,
+                             processor_meta, interpret_output):
```

Comment:
This function needs a docstring. Plus it doesn't use `crash_id`, so we can nix that argument.